Recon/src/Recon/Skims/EcalPreselectionSkimmer.cxx

```diff
@@ -0,0 +1,62 @@
+/**
+ * @file EcalPreselectionSkimmer.cxx
+ * @brief Processor used to pre-select events for the ECAL studies
+ * @author Tamas Almos Vami (UCSB)
+ */
+
+#include "Recon/Skims/EcalPreselectionSkimmer.h"
+
+namespace recon {
+
+EcalPreselectionSkimmer::EcalPreselectionSkimmer(const std::string &name,
+                                                 framework::Process &process)
+    : framework::Producer(name, process) {}
+
+void EcalPreselectionSkimmer::configure(framework::config::Parameters &ps) {
+  ecal_veto_name_ = ps.getParameter<std::string>("ecal_veto_name");
+  ecal_veto_pass_ = ps.getParameter<std::string>("ecal_veto_pass");
+  summed_det_max_ = ps.getParameter<double>("summed_det_max");  // MeV
+  summed_tight_iso_max_ =
+      ps.getParameter<double>("summed_tight_iso_max");  // MeV
+  ecal_back_energy_max_ =
+      ps.getParameter<double>("ecal_back_energy_max");  // MeV
+  n_readout_hits_max_ = ps.getParameter<int>("n_readout_hits_max");
+  shower_rms_max_ = ps.getParameter<int>("shower_rms_max");
+  shower_y_std_max_ = ps.getParameter<int>("shower_y_std_max");
+  shower_x_std_max_ = ps.getParameter<int>("shower_x_std_max");
+  max_cell_dep_max_ = ps.getParameter<double>("max_cell_dep_max");  // MeV
+  std_layer_hit_max_ = ps.getParameter<int>("std_layer_hit_max");
+  n_straight_tracks_max_ = ps.getParameter<int>("n_straight_tracks_max");
+
+  return;
+}
+
+void EcalPreselectionSkimmer::produce(framework::Event &event) {
+  bool passedPreselection{false};
+  auto ecalVeto{
```

Comment:
```suggestion
  const auto& ecalVeto{
```

Reply:
done now. Also FYI clang like to put the `&` exactly next to the variable, i.e.`const auto &ecalVeto{`